test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -1394,6 +1465,22 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
             expect(test.form.securityBeacon().css('background-image')).toMatch(/\/base\/target\/img\/security\/default.*\.png/);
           });
       });
+      itp('shows beacon-loading animation while loading security image (with deviceFingerprint)', function () {
+        return setup({ features: { securityImage: true, deviceFingerprinting: true }})
+          .then(function (test) {
+            test.securityBeacon = test.router.header.currentBeacon.$el;
+            spyOn(test.securityBeacon, 'toggleClass');
+            test.setNextResponse(resSecurityImage);
```

Comment:
Dont we show loading when we make call to get the actual image. 

I would think that we make dfp call - toggle bacon loading state  
once dfp call is back get make getImage call - toggle bacon loading state

Reply:
we don't for the getImage request, we only do for device fingerprint
so this test is to know the beacon changed to loading and back. This happens during the deviceFingerprint call.

I guess we could improve this even more by adding loading animation while getImage request is made too. I'll file a follow up jira

Reply:
cool might be tricky to do that since we dont do an api call it is just a get image call